collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -237,25 +237,46 @@ static String toSnakeAndLowerCase(String attrName) {
       return resultBuilder.toString();
     }
 
+  /**
+   * Change invalid chars to underscore, and merge underscores.
+   * @param name Input string
+   * @return
+   */
+  static String safeName(String name) {
+      if (name == null) {
+        return null;
+      }
+      boolean prevCharIsUnderscore = false;
+      StringBuilder safeNameBuilder = new StringBuilder();
```

Comment:
You can provide a capacity here, it'll probably be a little more efficient

Reply:
Filled in capacity here and for the `toSnakeAndLowerCase` method above.